File: unsorted_code/oracle_connect_test.py
import cx_Oracle

def getQuery(oracleConn):
    oracleCursor = oracleConn.cursor()
    oracleCursor.execute("SELECT USERNAME FROM all_users")
    orclUserNames = [x[0] for x in oracleCursor.fetchall()]
    oracleCursor.close()
    oracleConn.close()
    return orclUserNames

if __name__== "__main__":
    dsn_tns = '(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=jamalvm01.ddns.net)(PORT=1539))(CONNECT_DATA=(SID=orcl)))'
    test_local = True
    oracleConn = cx_Oracle.connect('webdbtool', 'f15cim2w3e4r', dsn_tns)
    print(getQuery(oracleConn))

    if test_local:
        dsn_tns = '(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=jamalvm01.ddns.net)(PORT=1539))(CONNECT_DATA=(SID=orcl)))'
        oracleConn = cx_Oracle.connect('webdbtool', 'f15cim2w3e4r', dsn_tns)
        print(getQuery(oracleConn))

File: unsorted_code/oracle_connect_test.R
Sys.setenv(PATH = paste0(Sys.getenv("PATH"), ";C:\\Oracle\\instantclient_12_2"))
library(ROracle)
library(data.table)
library(pipeR)
Sys.setenv(TZ = "Asia/Taipei", ORA_SDTZ = "Asia/Taipei")

# local
host <- "192.168.1.113"
port <- 1539
sid <- "orcl"
connectString <- sprintf("(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=%s)(PORT=%i))(CONNECT_DATA=(SID=%s)))",
                         host, port, sid)
oraConn <- dbConnect(dbDriver("Oracle"), username = "webdbtool",
                     password = "f15cim2w3e4r", dbname = connectString)
dbListTables(oraConn)
dbDisconnect(oraConn)

# remote
host <- "jamalvm01.ddns.net"
port <- 15039
sid <- "orcl"
connectString <- sprintf("(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=%s)(PORT=%i))(CONNECT_DATA=(SID=%s)))",
                         host, port, sid)
oraConn <- dbConnect(dbDriver("Oracle"), username = "webdbtool",
                     password = "f15cim2w3e4r", dbname = connectString)
dbListTables(oraConn)
dbDisconnect(oraConn)


